### PR TITLE
Add MMXLOG_REMOVE_HEAP_FILE and MMXLOG_REMOVE_APPENDONLY_FILE xlog records.

### PR DESCRIPTION
**Pull request by asubramanya**

### CODE REVIEW

File: src/include/cdb/cdbpersistentfilesysobj.h [hunk 1]
```diff
@@ -163,6 +164,7 @@ extern PersistentFileSysObjStateChangeResult PersistentFileSysObj_StateChange(
 	bool						retryPossible,
 	bool						flushToXLog,
 	PersistentFileSysState		*oldState,
+	PersistentFileSysRelStorageMgr relStorageMgr,
```
Comment (ashwinstar) [inline src/include/cdb/cdbpersistentfilesysobj.h]: It seems most of the function definitions were changed to pass the `relStorageMgr` to this function `PersistentFileSysObj_StateChange()`. Seems inside this function we do `PersistentFileSysObj_ReadTuple()`, so this tuple should have info needed and avoid passing the `relStorageMgr` to this function. That should avoid lot of changes ?

Comment (asimrp) [reply src/include/cdb/cdbpersistentfilesysobj.h]: +1

**Comment by asubramanya** [comment]
@asimrp @ashwinstar Addressed PR comments. Please take a look.

File: src/backend/cdb/cdbpersistentfilesysobj.c [hunk 2]
```diff
@@ -1264,6 +1264,8 @@ PersistentFileSysObjStateChangeResult PersistentFileSysObj_StateChange(
 							values,
 							&tupleCopy);
 
+	PersistentFileSysRelStorageMgr relStorageMgr = (PersistentFileSysRelStorageMgr)DatumGetInt16(values[Anum_gp_persistent_relation_node_relation_storage_manager - 1]);
```
Comment (ashwinstar) [inline src/backend/cdb/cdbpersistentfilesysobj.c]: Need to be under If `fsObjType == PersistentFsObjType_RelationFile` ?

Comment (asubramanya) [reply src/backend/cdb/cdbpersistentfilesysobj.c]: Yup. Updated.

File: src/backend/cdb/cdbpersistenttablespace.c [hunk 3]
```diff
@@ -1244,7 +1244,8 @@ PersistentTablespace_DroppedVerifiedActionCallback(
 												   int64 persistentSerialNum,
  /* Serial number for the relation.	Distinquishes the uses of the tuple. */
 
-												   PersistentFileSysObjVerifyExpectedResult verifyExpectedResult)
+												   PersistentFileSysObjVerifyExpectedResult verifyExpectedResult,
+												   PersistentFileSysRelStorageMgr relStorageMgr)
```
Comment (asimrp) [inline src/backend/cdb/cdbpersistenttablespace.c]: Doesn't this unused argument cause a compiler warning?

Comment (asubramanya) [reply src/backend/cdb/cdbpersistenttablespace.c]: It doesn't. There are other unused arguments to this function e.g persistentTid and the compiler doesn't seem to complain about them as well.

**Pull request merged**